Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Storage/Internal/FbSqlGenerationHelper.cs

```diff
@@ -15,14 +15,54 @@
 
 //$Authors = Jiri Cincura (jiri@cincura.net), Jean Ressouche, Rafael Almeida (ralms@ralms.net)
 
+using System;
+using System.Text; 
+using FirebirdSql.EntityFrameworkCore.Firebird.Extensions;
+using FirebirdSql.EntityFrameworkCore.Firebird.Infrastructure.Internal;
 using Microsoft.EntityFrameworkCore.Storage;
 
 namespace FirebirdSql.EntityFrameworkCore.Firebird.Storage.Internal
 {
 	public class FbSqlGenerationHelper : RelationalSqlGenerationHelper
 	{
-		public FbSqlGenerationHelper(RelationalSqlGenerationHelperDependencies dependencies)
+		private readonly IFbOptions _options;
+
+		public FbSqlGenerationHelper(RelationalSqlGenerationHelperDependencies dependencies, IFbOptions options)
 			: base(dependencies)
-		{ }
+		{
+			_options = options;
+		}
+
+		public override string EscapeIdentifier(string identifier)
+		{
+			return identifier.MaxLength(_options.Settings.ObjectLengthName);
```

Comment:
I think this isn't correct. It might collide with other entities/objects or with other objects in database. Best we can do is to throw an exception.

Reply:
Truth! I'll do it!

Reply:
In our code we made hashing of names that don't fit in 31 symbols restrictions. Looks ugly at database view but without collisions (though not guaranteed, but still). I'd prefer to have my custom implementation for this, so I could configure Escaping from our side.